dev/mcal/inc/stm32f103xx.h

```diff
@@ -82,6 +82,79 @@ struct RccRegDef{
 #define RCC  (reinterpret_cast<volatile RccRegDef*>(RCC_BASE_ADDRESS))
 
 }  // namespace rcc
+
+namespace gpio {
+struct GpioRegDef {
+    RegWidth_t CRL;     // Port configuration register low
+    RegWidth_t CRH;     // Port configuration register High
+    RegWidth_t IDR;
+    RegWidth_t ODR;
+    RegWidth_t BSRR;
+    RegWidth_t BRR;
+    RegWidth_t LCKR;
+};
+
+#define GPIOA  (reinterpret_cast<volatile GpioRegDef*>(GPIOA_BASE_ADDRESS))
+#define GPIOB  (reinterpret_cast<volatile GpioRegDef*>(GPIOB_BASE_ADDRESS))
+#define GPIOC  (reinterpret_cast<volatile GpioRegDef*>(GPIOC_BASE_ADDRESS))
+
+}  // namespace gpio
+
+namespace afio {
+struct AfioRegDef {
+    union EVCR {
+        struct {
+            RegWidth_t PIN      :4;
+            RegWidth_t PORT     :3;
+            RegWidth_t EVOE     :1;
+            RegWidth_t          :24;        // Reserved
+        };
+        RegWidth_t registerVal;
+    }EVCR;
+    union MAPR {
+        struct {
+            RegWidth_t SPI1_REMAP           :1;
```

Comment:
Also, you may add a comment here for each member of the struct to describe its functionality